packages/app/src/pages/SwapPage/SwapComponent.tsx

```diff
@@ -40,20 +40,22 @@ export function SwapComponent({
 
   const handleInvertCoins = () => {
     if (activeInput.current === ActiveInput.to) {
-      activeInput.current = ActiveInput.from;
+      const from = fromInput.amount;
       fromInput.setAmount(toInput.amount);
-      toInput.setAmount(null);
+      toInput.setAmount(from);
     } else {
-      activeInput.current = ActiveInput.to;
```

Comment:
This is causing the issue you see when inverting fields

Reply:
We use reference here because we need a nonstate delayed change.

Reply:
https://user-images.githubusercontent.com/3941923/169627280-200c409c-6db0-4329-8632-cc8b4b5aa736.mov

Reply:
Thanks for adding the `startTransition`